kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/LogSegment.java

```diff
@@ -149,6 +162,28 @@ public LogSegment(
                 String.format("Delta versions must be contiguous: %s", deltaVersions));
           }
         }
+
+        // Check that our compactions are in range
+        checkArgument(
+            compactions.stream()
+                .allMatch(
+                    fs -> {
+                      Tuple2<Long, Long> versions = FileNames.logCompactionVersions(fs.getPath());
+                      return versions._2 <= version;
+                    }),
+            "compactions must have end version <= version");
```

Comment:
Can this be moved up next to the `compactions must have start version less than end version` check?